### PR TITLE
COMPAT: some compatibility fixes with new numpies

### PR DESCRIPTION
**Pull request by jreback**

### CODE REVIEW

File: pandas/io/tests/test_pytables.py [hunk 1]
```diff
@@ -848,11 +848,11 @@ def test_append(self):
...
             uint_data = DataFrame({
-                'u08': Series(np.random.random_integers(0, high=255, size=5),
+                'u08': Series(np.random.randint(0, high=255, size=5),
                               dtype=np.uint8),
-                'u16': Series(np.random.random_integers(0, high=65535, size=5),
+                'u16': Series(np.random.randint(0, high=65535, size=5),
                               dtype=np.uint16),
-                'u32': Series(np.random.random_integers(0, high=2**30, size=5),
+                'u32': Series(np.random.randint(0, high=2**30, size=5),
                               dtype=np.uint32),
```
Comment (gfyoung) [inline pandas/io/tests/test_pytables.py]: Not sure if this a problem, but these function calls aren't quite equivalent (closed vs. open interval for `random_integers` and `randint` respectively).

File: pandas/computation/tests/test_eval.py [hunk 2]
```diff
@@ -50,13 +50,7 @@ def _eval_single_bin(lhs, cmp1, rhs, engine):
...
             return c(lhs, rhs)
         except ValueError as e:
-            try:
-                msg = e.message
-            except AttributeError:
-                msg = e
-            msg = u(msg)
-            if msg == u('negative number cannot be raised to a fractional'
-                        ' power'):
+            if str(e).startswith('negative number cannot be raised to a fractional power'):
```
Comment (gfyoung) [inline pandas/computation/tests/test_eval.py]: You're missing one more in this file.  Just search for `msg = e.message` (deprecated) in your changed file and get rid of them.

File: pandas/io/tests/test_pytables.py (hunk 1 above)
Comment (jreback) [inline pandas/io/tests/test_pytables.py]: they seem the same.

File: pandas/computation/tests/test_eval.py (hunk 2 above)
Comment (jreback) [inline pandas/computation/tests/test_eval.py]: hmm

File: pandas/io/tests/test_pytables.py (hunk 1 above)
Comment (gfyoung) [inline pandas/io/tests/test_pytables.py]: I mean it's a moot point if Travis is still happy, but for reference, they are not by definition.

**Comment by jreback** [comment]
@tacaswell @mdboom, using mpl 1.5.0 here. I was turning on deprecation warnings. (we get a bunch of these in test suite), known issue?

cc @TomAugspurger 
```
======================================================================
ERROR: test_ts_area_lim (pandas.tests.test_graphics.TestSeriesPlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "pandas/tests/test_graphics.py", line 612, in test_ts_area_lim
    ax = self.ts.plot.area(stacked=False, x_compat=True)
  File "pandas/tools/plotting.py", line 3688, in area
    return self(kind='area', **kwds)
  File "pandas/tools/plotting.py", line 3564, in __call__
    **kwds)
  File "pandas/tools/plotting.py", line 2643, in plot_series
    **kwds)
  File "pandas/tools/plotting.py", line 2439, in _plot
    plot_obj.generate()
  File "pandas/tools/plotting.py", line 1032, in generate
    self._post_plot_logic_common(ax, self.data)
  File "pandas/tools/plotting.py", line 1160, in _post_plot_logic_common
    fontsize=self.fontsize)
  File "pandas/tools/plotting.py", line 1208, in _apply_axis_properties
    labels = axis.get_majorticklabels() + axis.get_minorticklabels()
  File "/Users/jreback/miniconda/lib/python2.7/site-packages/matplotlib/axis.py", line 1161, in get_majorticklabels
    ticks = self.get_major_ticks()
  File "/Users/jreback/miniconda/lib/python2.7/site-packages/matplotlib/axis.py", line 1290, in get_major_ticks
    numticks = len(self.get_major_locator()())
  File "/Users/jreback/miniconda/lib/python2.7/site-packages/matplotlib/dates.py", line 1007, in __call__
    return self._locator()
  File "/Users/jreback/miniconda/lib/python2.7/site-packages/matplotlib/dates.py", line 813, in __call__
    return self.tick_values(dmin, dmax)
  File "/Users/jreback/miniconda/lib/python2.7/site-packages/matplotlib/dates.py", line 830, in tick_values
    self.rule.set(dtstart=start, until=stop, count=self.MAXTICKS + 1)
  File "/Users/jreback/miniconda/lib/python2.7/site-packages/matplotlib/dates.py", line 724, in set
    self._rrule = rrule(**self._construct)
  File "/Users/jreback/miniconda/lib/python2.7/site-packages/dateutil/rrule.py", line 470, in __init__
    "raise an error.", DeprecationWarning)
DeprecationWarning: Using both 'count' and 'until' is inconsistent with RFC 2445 and has been deprecated in dateutil. Future versions will raise an error.
```

**Comment by mdboom** [comment]
Thanks for pointing out.  Not a known issue (to me at least).  I'll see if I can reproduce locally.

**Comment by jreback** [comment]
@mdboom this is using pretty recently released ``python-dateutil 2.5.3`` FYI (might be new there)

**Comment by tacaswell** [comment]
There is a pr fixing this, but I am failing to find it.

**Comment by tacaswell** [comment]
https://github.com/matplotlib/matplotlib/issues/6155 / https://github.com/matplotlib/matplotlib/pull/6190

**Comment by mdboom** [comment]
Thanks @tacaswell.  Unfortunately, it doesn't look there's much of a way to workaround that from the outside...

**Comment by tacaswell** [comment]
the fix will be it 1.5.2 / 2.0, which ever comes first

**Comment by jreback** [comment]
thanks @mdboom @tacaswell no worries. just trying to have clean test outputs.

**Pull request merged**